iothub_client/samples/iothub_client_device_cbor_sample/CMakeLists.txt

```
@@ -0,0 +1,51 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+if(NOT ${use_mqtt})
+    message(FATAL_ERROR "iothub_client_device_cbor_sample being generated without MQTT support")
+endif()
+
+compileAsC99()
+
+IF(WIN32)
+    #windows needs this define
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+    add_definitions(-DGB_MEASURE_MEMORY_FOR_THIS -DGB_DEBUG_ALLOC)
+ENDIF(WIN32)
+
+if(MSVC)
+  set(CMAKE_EXE_LINKER_FLAGS
+      "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:LIBCMT.lib") # To compile with intel/tinycbor
+endif()
+
+# Sample Executable
+set(iothub_client_device_cbor_sample_c_files
+    iothub_client_device_cbor_sample.c
+)
+
+if(${use_sample_trusted_cert})
+    add_definitions(-DSET_TRUSTED_CERT_IN_SAMPLES)
+    include_directories(${PROJECT_SOURCE_DIR}/certs)
+    set(iothub_client_device_cbor_sample_c_files ${iothub_client_device_cbor_sample_c_files}
+        ${PROJECT_SOURCE_DIR}/certs/certs.c)
+endif()
+
+add_executable(iothub_client_device_cbor_sample
+    ${iothub_client_device_cbor_sample_c_files}
```

can we just have the `.c` file here? I don't think this indirection is adding anything
```suggestion
    iothub_client_device_cbor_sample.c
```

---

We need a way to add the certs.c file if that option is selected.  Also, this follows the same approach of the other sample's cmakes to add the optional certs.c.

---

yup this can still be done with the below code snipped:
```
target_sources(iothub_client_device_cbor_sample PRIVATE
         ${PROJECT_SOURCE_DIR}/certs/certs.c
```